ts/private/ts_project_worker.js

```diff
@@ -419,41 +584,65 @@ function createProgram(args, initialInputs) {
         noop,
         noop
     );
+    // deleting this will make tsc to not schedule updates but wait getProgram to be called to apply updates which is exactly what is needed.
+    delete host.setTimeout;
+    delete host.clearTimeout;
```

Comment:
nit: create issue to test against multiple typescript versions since we don't know what the minimum version is for worker support

Reply:
done https://github.com/aspect-build/rules_ts/issues/295